DAG deploy updates (#2872)
astronomer/docs: astro/deploy-dags.md

DAG-only deploys are the fastest way to deploy code to Astro. They are recommended if you only need to deploy changes made to the `dags` directory of your Astro project.

> DAG-only deploys are enabled by default on all Deployments on Astro Hosted. After it is enabled, you must still [deploy your project image](deploy-project-image.md) when you make a change to any file in your Astro project that is not in the `dags` directory and for [runtime upgrades](upgrade-runtime.md).

Author: ```suggestion
DAG-only deploys are enabled by default on all Deployments on Astro Hosted. When they are enabled, you must still [deploy your project image](deploy-project-image.md) when you make a change to any file in your Astro project that is not in the `dags` directory, or when you [upgrade Astro Runtime](upgrade-runtime.md).
```

astro deploy --dags
```

> ## Enable/ disable DAG-only deploys on a Deployment

Author: According to google style guide, we should be avoiding slashes altogether! Making that update now.

As a Workspace Owner, you can enable or disable DAG-only deploys for a Deployment. On Astro Hosted, DAG-only deploys are enabled by default for all Deployments. On Astro Hybrid, you must enable them manually for each Deployment. 
  
To determine if DAG-only deploy functionality is the right choice for your organization, contact [Astronomer support](https://cloud.astronomer.io/support). 

:::warning

Carefully read and complete all of the following steps to ensure that disabling or enabling DAG-only deploys doesn't disrupt your Deployment. **Immediately after you update the setting, you must trigger an image deploy to your Astro Deployment using `astro deploy`.** If you don't complete this step, you can't access your DAGs in the Airflow UI.

:::

### Prerequisites

Before you enable or disable DAG-only deploys on a Deployment, ensure the following:

- You have Workspace Owner permissions for the Workspace that hosts the Deployment. 
- You have access to the latest version of your Deployment's Astro project.
- You can update your Deployment using the Astro CLI. 

### Enable DAG-only deploys

> After you enable DAG-only deploys:

Author: ```suggestion
DAG-only deploys are enabled by default on Astro Hosted. You only need to enable DAG-only deploys if you're on Astro Hybrid or if you've previously disabled them on a Hosted Deployment. After you enable DAG-only deploys:
```

- You can run `astro deploy --dags` to deploy only DAGs to your Deployment.
- In the Cloud UI, your Deployment **DAG bundle version** updates when you trigger an image deploy or a DAG-only deploy. 
- Your Deployment includes infrastructure for deploying your DAGs separately from your project image. See [What happens during a code deploy](deploy-project-image.md#what-happens-during-a-project-deploy).

> 1. Run the following command to enable DAG-only deploys:

Reviewer: Will this actually be an optional step if they're enabled by default for Astro Hosted?

Author: Good point, but I think I'll make that note before the steps even start.

    ```sh
    astro deployment update --dag-deploy enable
    ```

2. Run the following command to deploy all of the files in your Astro project as a Docker image:

    ```sh
    astro deploy
    ```

### Disable DAG-only deploys 

After you disable DAG-only deploys:

> - You can no longer run `astro deploy --dags` to trigger a DAG-only deploy to your Deployment.

Reviewer: ```suggestion
- You can't run `astro deploy --dags` to trigger a DAG-only deploy to your Deployment.
```

- Any changes to your DAG code are deployed as part of your Astro project Docker image.
- In the Cloud UI, your Deployment **DAG bundle version** doesn't update when you deploy code. 

> - Your Deployment no longer includes infrastructure for deploying your DAGs separately from your project image. See [What happens during a code deploy](deploy-project-image.md#what-happens-during-a-project-deploy).

Reviewer: ```suggestion
- Your Deployment doesn't include infrastructure for deploying your DAGs separately from your project image. See [What happens during a code deploy](deploy-project-image.md#what-happens-during-a-project-deploy).
```

Have some concerns about the "no longer" being a little difficult to parse for anyone reading quickly or isn't super comfortable working in English

astro deploy --dags
```

> ## Toggle DAG-only deploys on a Deployment

Reviewer: I know we had some discussion about this phrasing, so I looked up **Toggle** in the google style guide to see how other folks are handling it, and it really does seem like it's a term that's only specifically used to refer to a Toggle UI element. 

Enable/disable and Turn on/off were their recommended alternatives but not sure if it's worth a long discussion on which way we go.

Author: I liked Enable/ disable to begin with, so I'm happy to revert here! I think there was a previous review comment or discussion on Zoom that suggested other wording, but I def agree it makes sense to remove the UI-based "toggle" language

As a Workspace Owner, you can enable or disable DAG-only deploys for a Deployment. On Astro Hosted, DAG-only deploys are enabled by default for all Deployments. On Astro Hybrid, you must enable them manually for each Deployment. 
  
To determine if DAG-only deploy functionality is the right choice for your organization, contact [Astronomer support](https://cloud.astronomer.io/support). 

### Prerequisites

Before you enable or disable DAG-only deploys on a Deployment, ensure the following:

- You have Workspace Owner permissions for the Workspace that hosts the Deployment. 
- You have access to the latest version of your Deployment's Astro project.
- You can update your Deployment using the Astro CLI. 

### Enable DAG-only deploys

:::warning

> Carefully read and complete all of the following steps to ensure that your Deployment is not disrupted by disabling or enabling DAG-only deploys. Crucially, you must trigger an image deploy to your Astro Deployment using `astro deploy` immediately after you apply the setting. If you don't, your DAGs will be not be available in the Airflow UI until you update your Deployment.

Reviewer: ```suggestion
Carefully read and complete all of the following steps to ensure that your Deployment is not disrupted by enabling DAG-only deploys. After enabling DAG-only deploys, you must immediately trigger an image deploy to your Astro Deployment using `astro deploy`. After updating the complete image, you can access your DAGs in the Airflow UI and choose to update the full image or just DAGs when deploying to Astro.
```
I think this note makes sense if it appears at the VERY beginning of this section, but with two notes placed at each of the enable/disable, i'm worried the boilerplate will be too easy to skip.

Author: Agreed on note placement. Changed the copy here a bit to still have a scary "if" statement at the end which explains what happens when users don't complete the setup.

:::

When you enable DAG-only deploys and trigger a code deploy: 

- You can run `astro deploy --dags` to deploy only DAGs to your Deployment.

> - In the Cloud UI, your Deployment **DAG bundle version** updates when you trigger both image deploys and DAG-only deploys. 

Reviewer: Is this a both...and or an either...or?

So the current phrasing, the DAG bundle version updates when [image+DAGs] are deployed. 

If the DAG bundle version updates when either the image deploys OR with a DAG only Deploy, we might need to change it.

> - In the Cloud UI, your Deployment **DAG bundle version** updates when you trigger both image deploys and DAG-only deploys. 

Reviewer: ```suggestion
- In the Cloud UI, your Deployment **DAG bundle version** updates when you trigger either image deploys or DAG-only deploys. 
```

- Your Deployment includes infrastructure for deploying your DAGs separately from your project image. See [What happens during a code deploy](deploy-project-image.md#what-happens-during-a-project-deploy).

1. Run the following command to enable DAG-only deploys:

    ```sh
    astro deployment update --dag-deploy enable
    ```

2. Run the following command to deploy all of the files in your Astro project as a Docker image:

    ```sh
    astro deploy
    ```

### Disable DAG-only deploys 

:::warning

> Carefully read and complete all of the following steps to ensure that your Deployment is not disrupted by disabling or enabling DAG-only deploys. Crucially, you must trigger an image deploy to your Astro Deployment using `astro deploy` immediately after you apply the setting. If you don't, your DAGs will be not be available in the Airflow UI until you update your Deployment.

Reviewer: ```suggestion
Carefully read and complete all of the following steps to ensure that disabling or enabling DAG-only deploys doesn't disrupt your Deployment. **Immediately after you update the setting, you must trigger an image deploy to your Astro Deployment using `astro deploy`.** You can't access your DAGs in the Airflow UI until you update your Deployment with an image deploy.
```

There was some passive voice in this note that made it feel a little boilerplate-y. I know we have a similar note in the preceding section, but there's a real risk of when warnings seem too repetitive or similar, people just skip them.
